docs/Datagrid.md

```diff
@@ -512,27 +539,28 @@ By default, `<Datagrid>` renders the table head using `<DatagridHeader>`, an int
 
 For instance, here is a simple datagrid header that displays column names with no sort and no "select all" button:
 
-```jsx
-import { TableHead, TableRow, TableCell } from '@mui/material';
+```tsx
+import * as React from "react";
+import { TableHead, TableRow, TableCell } from "@mui/material";
+import { DatagridHeaderProps, FieldProps, List, Datagrid, DatagridHeader } from "react-admin";
 
-const DatagridHeader = ({ children }) => (
+const DatagridHeader = ({ children }: DatagridHeaderProps) => (
```

Comment:
In this snippet you are both importing `DatagridHeader` from `"react-admin"` and declaring `const DatagridHeader`. This cannot work. You should remove the import I believe.